### PR TITLE
API Functional Tests EUI-1592

### PR DESCRIPTION
**Pull request by Vamshi-Muniganti**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: test/integration/tests/post_Invite_User.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    lastName: 'Muniganti',
+    email: 'vam.mun1752@mailnesia.com',
+    roles: ['pui-organisation-manager'],
+    resendInvite: true,
+  };
+  test('POST Invite User', () => generatePOSTAPIRequest ('POST', '/api/reinviteUser?organisationId=2GIHJH9', payload)
+     // console.log('response', response.headers.get('cache-control'))
+        .then(response => {
+          if (response.status === 429 ) {
+          console.log(`User Already Invited: ${response.status}`);
```
Comment (sreekanthpuligadda) [inline test/integration/tests/post_Invite_User.ts:19]: missing assertions statements. please add expect or assert statements .

File: test/integration/tests/post_Update_PBA.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { generatePOSTAPIRequest } from './utils';
+const should = require('chai').should();
+
+suite('Approve Org -> PUT Update PBA numbers', function() {
+  this.timeout(50000);
+  const payload = {
+  paymentAccounts: [`PBA1${Math.floor(100000 + Math.random() * 900000)}`, `PBA2${Math.floor(100000 + Math.random() * 900000)}`],
+  orgId: 'BOHOBFK',
+  };
+  test('PUT Update PBA numbers', () => generatePOSTAPIRequest ('PUT', '/api/updatePba', payload));
```
Comment (sreekanthpuligadda) [inline test/integration/tests/post_Update_PBA.ts:10]: missing assertion statement in test.

File: test/integration/tests/post_Invite_User.ts (hunk 1 above)
Comment (Vamshi-Muniganti) [reply test/integration/tests/post_Invite_User.ts:19]: dont need assertion as its tricky to validate the email id which activates every hour, so basically verifying the response codes and it throws any exception in fetch

File: test/integration/tests/post_Update_PBA.ts (hunk 2 above)
Comment (Vamshi-Muniganti) [reply test/integration/tests/post_Update_PBA.ts:10]: not required for PBA as they randomly generated jsut validating the request

File: test/integration/config/environments/sprod.config.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+export default {
+    services: {
```
Comment (udaydenduluri33) [inline test/integration/config/environments/sprod.config.ts:2]: if not using please delete this file

File: test/integration/tests/Approve_Pending_Organisations.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+let org = '';
+let body = '';
+suite('Approve Org -> Get Pending Organisation details', function() {
+  this.timeout(50000);
+  test('GET Pending Organisations details', () => generateAPIRequest ('GET', '/api/organisations?status=PENDING')
+     // console.log('response', response.headers.get('cache-control'))
+        .then(response => {
+          response.status.should.be.eql(200);
+          response.add = {jurisdictions: [ {id : 'SSCS'}, {id: 'AUTOTEST1'}, {id: 'DIVORCE'}, {id: 'PROBATE'}]},
+          org = response.data[0].organisationIdentifier;
```
Comment (udaydenduluri33) [inline test/integration/tests/Approve_Pending_Organisations.ts:13]: Do we need to check if `response.data && response.data.length > 0`

File: test/integration/tests/Delete_Active_Organisations.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+const should = require('chai').should();
+let org = '';
+
+suite('Approve Org -> Get Active Organisation details', function() {
+  this.timeout(50000);
+  test('GET Pending Organisations details', () => generateAPIRequest ('GET', '/api/organisations?status=ACTIVE')
+     // console.log('response', response.headers.get('cache-control'))
+        .then(response => {
+          response.status.should.be.eql(200);
+          org = response.data[0].organisationIdentifier;
```
Comment (udaydenduluri33) [inline test/integration/tests/Delete_Active_Organisations.ts:12]: Do we need to check if response.data && response.data.length > 0

File: test/integration/tests/Delete_Pending_Organisations.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+const should = require('chai').should();
+let org = '';
+
+suite('Approve Org -> Get Pending Organisation details', function() {
+  this.timeout(50000);
+  test('GET Pending Organisations details', () => generateAPIRequest ('GET', '/api/organisations?status=PENDING')
+     // console.log('response', response.headers.get('cache-control'))
+        .then(response => {
+          response.status.should.be.eql(200);
+          org = response.data[0].organisationIdentifier;
```
Comment (udaydenduluri33) [inline test/integration/tests/Delete_Pending_Organisations.ts:12]: Do we need to check if response.data && response.data.length > 0

File: test/integration/tests/get_Organisation_User_Details.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { generateAPIRequest } from './utils';
+const should = require('chai').should()
+
+suite('Approve Org -> Get Organisation Users details', function() {
+  this.timeout(50000);
+  test('GET Organisation User details', () => generateAPIRequest ('GET', '/api/organisations?usersOrgId=2GIHJH9')
+     // console.log('response', response.headers.get('cache-control'))
+        .then(response => {
+           response.status.should.be.eql(200);
+           console.log(response.data.users[0].email);
```
Comment (udaydenduluri33) [inline test/integration/tests/get_Organisation_User_Details.ts:10]: Do we need to check if r`esponse.data.users && response.data.users.length > 0`

File: test/integration/tests/utils/getCookie.ts [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    throw error;
+  }
+
+  const cookies: [] = await page.cookies();
+
+  let roles = '';
+  let webappCookie = '';
+
+  cookies.forEach((cookie: any) => {
+    if (cookie.name === 'XSRF-TOKEN') {
```
Comment (udaydenduluri33) [inline test/integration/tests/utils/getCookie.ts:35]: Try using switch-case instead of if here.

**Review by udaydenduluri33 — APPROVED** [review]
Discussed with @Vamshi-Muniganti  and agreed that these will be refactored in next PR. Thanks.

**Review by sreekanthpuligadda — APPROVED** [review]

**Pull request merged**